ql/src/java/org/apache/hadoop/hive/ql/parse/repl/CopyUtils.java

```diff
@@ -156,16 +163,11 @@ private void doCopyRetry(FileSystem sourceFs, List<ReplChangeManager.FileInfo> s
             srcFile.setIsUseSourcePath(false);
           }
         } else {
+          // The file copy might have failed for network issue. So retry with cm path again if the
+          // original file is missing.
           if (srcFile.isUseSourcePath()) {
             // Source file missing, then try with CM path
             srcFile.setIsUseSourcePath(false);
-          } else {
```

Comment:
Shall check if CM path exist and if not, shall throw the error/exception as in base code.